includes/class-wp-job-manager-email-notifications.php

```diff
@@ -0,0 +1,334 @@
+			if (
+				isset( $email_notifications[ $email_notification_key ] )
+				|| ( $enabled_notifications_only && ! self::is_email_notification_enabled( $email_notification_key ) )
+			) {
+				continue;
+			}
+
+			$email_notifications[ $email_notification_key ] = $email_class;
+		}
+
+		return $email_notifications;
+	}
+
+	/**
+	 * Generate the file name for the email template.
+	 *
+	 * @param string $template_name
+	 * @param bool   $plain_text
+	 * @return string
+	 */
+	public static function get_template_file_name( $template_name, $plain_text = false ) {
+		$file_name_parts = array( 'emails' );
+		if ( $plain_text ) {
+			$file_name_parts[] = 'plain';
+		}
+
+		$file_name_parts[] = $template_name . '.php';
+
+		return implode( '/', $file_name_parts );
+	}
+
+	/**
+	 * Returns the total number of deferred notifications to be sent.
+	 *
+	 * @access private
+	 *
+	 * @return int
+	 */
+	public static function _get_deferred_notification_count() {
+		return count( self::$deferred_notifications );
+	}
+
+	/**
+	 * Confirms an email notification is valid.
+	 *
+	 * @access private
+	 *
+	 * @param string $email_class
+	 * @return bool
+	 */
+	private static function is_email_notification_valid( $email_class ) {
+		// PHP 5.2: Using `call_user_func()` but `$email_class::get_key()` preferred.
+		return is_string( $email_class )
+				&& class_exists( $email_class )
+				&& is_subclass_of( $email_class, 'WP_Job_Manager_Email' )
+				&& false !== call_user_func( array( $email_class, 'get_key') )
+				&& false !== call_user_func( array( $email_class, 'get_name') );
+	}
+
+	/**
+	 * Sends an email notification.
+	 *
+	 * @access private
+	 *
+	 * @param string               $email_notification_key
+	 * @param WP_Job_Manager_Email $email
+	 * @return bool
+	 */
+	private static function send_email( $email_notification_key, WP_Job_Manager_Email $email ) {
+		if ( ! $email->is_valid() ) {
+			return false;
+		}
+
+		$fields = array( 'to', 'from', 'subject', 'rich_content', 'plain_content', 'attachments', 'cc', 'headers' );
+		$args = array();
+		foreach ( $fields as $field ) {
+			$method = 'get_' . $field;
+
+			/**
+			 * Filter email values for job manager notifications.
+			 *
+			 * @since 1.31.0
+			 *
+			 * @param mixed                $email_field_value Value to be filtered.
+			 * @param WP_Job_Manager_Email $email             Email notification object.
+			 */
+			$args[ $field ] = apply_filters( "job_manager_email_{$email_notification_key}_{$field}", $email->$method(), $email );
+		}
+
+		$headers = is_array( $args['headers'] ) ? $args['headers'] : array();
+
+		if ( ! empty( $args['from'] ) ) {
+			$headers[] = 'From: ' . $args['from'];
+		}
+
+		if ( ! self::send_as_plain_text( $email_notification_key ) ) {
+			$headers[] = 'Content-Type: text/html';
+		}
+
+		$content = self::get_email_content( $email_notification_key, $args );
+
+		return wp_mail( $args['to'], $args['subject'], $content, $headers, $args['attachments'] );
```

Comment:
Would be good to have a filter here for the ability to short circuit the email if needed.  Right now if I ever need to do this through `wp_mail` I have to short circuit by setting `to` and `message` to null/false values (until easy way to short circuit is integrated) https://core.trac.wordpress.org/ticket/35069

Reply:
eecfeee6d7fee77ac62c2f534d60201fdcfa24f6 work?